bin/thoth

```diff
@@ -1774,129 +1840,272 @@ handlers.load = function (client, argv)
+	env['MANTA_INPUT_FILE'] = dump_path;
+	env['MANTA_INPUT_OBJECT'] = info.dump;
+	env['THOTH_RUN_ANALYZER'] = (runAnalyzer ? 'true' : 'false');
 
-		status('debugging ' +
-		    mod_path.basename(mod_path.dirname(dump.dump)));
+	if (analyzer === undefined)
+		return (env);
 
-		var info = dump.name + '/info.json';
-		var cmd = 'echo "thoth: dump info is in \\$THOTH_INFO"\n';
+	if (analyzerIsDcmd(analyzer)) {
+		env['THOTH_ANALYZER_DCMD'] = analyzer;
+		return (env);
+	}
 
-		var args = [ dump.dump ];
+	env['THOTH_ANALYZER_NAME'] = analyzer;
+	env['THOTH_ANALYZER_OBJECT'] = mod_path.join(thoth.analyzers, analyzer);
 
-		var dummy = { phases: [ {} ] };
-		jobSetThoth(dummy);
-		cmd += dummy.phases[0].init + '\n';
+	return (env);
+}
 
-		if (analyzer) {
-			cmd += analyzerCmd(args);
-		} else {
-			/*
-			 * LIBPROC_INCORE_ELF=1 prevents us from loading
-			 * whatever node binary happens to be in $PATH. This is
-			 * necessary when debugging cores that do not match the
-			 * bitness of the node binary found in $PATH.
-			 */
-			cmd += 'LIBPROC_INCORE_ELF=1 mdb $MANTA_INPUT_FILE';
-		}
+var mget = function(client, path, mpath, cb)
+{
+	var write = mod_fs.createWriteStream(path);
 
-		setAnalyzer(client, cmd, function (asset) {
-			args = args.concat([ '-s', asset,
-			    '-s', dummy.phases[0].assets[0],
-			    '-c', '/assets/' + asset,
-			    '--memory=2048' ]);
+	client.info(mpath, function (err, info) {
+		checkError(err);
 
-			var child = mod_child.spawn(mlogin, args,
-			    { stdio: 'inherit' });
+		var read = client.createReadStream(mpath);
 
-			child.on('close', function (code) {
-				status('debugger exited with code ' + code);
+		progressBar(mpath, read, info.size);
 
-				client.unlink(asset, function (err) {
-					checkError(err);
-					process.exit(code);
-				});
-			});
-		}, { loadv8: analyzer ? false : true });
-	};
+		read.pipe(write);
 
-	if (argv.length > 1 && argv[argv.length - 1].indexOf('=') == -1)
-		analyzer = argv.pop();
+		write.on('finish', function (err) {
+			if (err) {
+				try {
+					mod_fs.unlinkSync(path);
+				} catch (_) {
+				}
 
-	if (argv.length == 1 && argv[0].indexOf('=') == -1) {
-		infoGet(client, argv[0], debug);
-	} else {
-		dumpsFromSpec(client, argv, { fields: [ 'dump' ] }, debug);
-	}
+				checkError(err);
+				cb(err);
+			}
+			cb(null, path);
+		});
+	});
 }
 
-handlers.index = function (client, argv)
+var downloadDump = function(client, info, cb)
 {
-	var scanComplete = false;
-	var id, time;
-	var seen = {};
+	var dump_path = mod_path.join('/var/tmp/thoth/cache', info.id,
+	    mod_path.basename(info.dump));
 
-	var job = { phases: [ {
-		type: 'map',
-		exec: 'DTRACE_DOF_INIT_DISABLE=1 ' +
-		    jobExecThoth('$MANTA_INPUT_FILE')
-	} ] };
+	if (mod_fs.existsSync(dump_path)) {
+		return (cb(null, dump_path));
+	}
+
+	status('downloading ' + mod_path.basename(info.dump) +
+	    ' to local cache');
+
+	mod_fs.mkdirSync(mod_path.dirname(dump_path), { recursive: true });
```

Comment:
That `recursive` doesn't work with node less than v10.  After hacking to use the non-jobby code path I hit this:

```
[root@5d4f7599-a991-6b35-dd44-d91936957a6b ~/joy/manta-thoth]# thoth debug f824873602b1c53d
thoth: downloading core.node.97708 to local cache
thoth: Error: ENOENT: no such file or directory, mkdir '/var/tmp/thoth/cache/f824873602b1c53dcec2a6daba6be3a5'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:932:18)
    at downloadDump (/root/joy/manta-thoth/bin/thoth:1925:9)
    at runDebugLocal (/root/joy/manta-thoth/bin/thoth:1942:2)
    at /root/joy/manta-thoth/bin/thoth:2050:5
    at /root/joy/manta-thoth/bin/thoth:248:3
    at /root/joy/manta-thoth/node_modules/manta/lib/client.js:1145:21
    at IncomingMessage.onEnd (/root/joy/manta-thoth/node_modules/manta/lib/client.js:492:13)
```

here is a patch to use the mkdirp package, if you want: https://gist.github.com/trentm/5ea375667a5fe319852d9d1193c478ad

Reply:
thanks